### ISSUE TITLE
[java] UnusedAssignment false-positive with try-with-resources

### ISSUE THREAD

**Issue by boris-petrov** [pmd/pmd#4127]
**Affects PMD Version:** 
6.49.0

**Rule:**
UnusedAssignment
https://pmd.sourceforge.io/pmd-6.49.0/pmd_rules_java_bestpractices.html#unusedassignment

**Description:**
Check the code sample

**Code Sample demonstrating the issue:**

```java
Lock lock;
synchronized (locks) {
  lock = locks.get(0);
}
try (lock) {
  ...
}
```

For some type `Lock` that is `AutoCloseable`.

**Expected outcome:**
PMD reports a violation at line `lock = locks.get(0);`, but that's wrong. That's a false positive.

**Running PMD through:** Gradle

**Comment by jsotuyod** [comment]
This is no longer reproducible under PMD 7.0.0
